feat: Support Cargo alternate registries in aas renovate (#18941)
renovatebot/renovate: docs/usage/rust.md

Renovate updates Rust crates by default.

## Private git-based crate registries / private Git dependencies

If any dependencies are hosted in private Git repositories, [Git Authentication for cargo](https://doc.rust-lang.org/cargo/appendix/git-authentication.html) must be set up.

If any dependencies are hosted on private crate registries (i.e., not on `crates.io`), then credentials should be set up in such a way that the Git command-line is able to clone the registry index.
Third-party crate registries usually provide instructions to achieve this.

Both of these are currently only possible when running Renovate self-hosted.

## Private sparse index registries

> If dependencies are available via a sparse index, then a `registryAlias` can be
> used to have Renovate accesss the dependencies via the sparse protocol. Use
> `HostRules` to provide authentication details if needed. Crates-io index access
> will still be via Renovates shared clone of the public git index.

Reviewer: ```suggestion
If dependencies are available via a "sparse index", you may use `registryAliases` to let Renovate accesses the dependencies via the `sparse` protocol.
If needed, set the authentication details with the `hostRules` config option.
`crates.io` index access will still be via Renovates shared clone of the public Git index.
```

Rewrite:
  - Use `you` to draw the reader into the text
  - Put "sparse index" between quotes, to highlight the term
  - Use simple words and phrases
  - One sentence per line
  - Use `crates.io` to match line 26 of this file
  
Grammar/typos:

- The `hostRules` config option name starts with a normal case latter
- The config option is `registryAliases` not `registryAlias`
- Git is a proper noun, so must be capitalized

Renovate updates Rust crates by default.

## Private Git-based crate registries / private Git dependencies

If any dependencies are hosted in private Git repositories, [Git Authentication for cargo](https://doc.rust-lang.org/cargo/appendix/git-authentication.html) must be set up.

If any dependencies are hosted on private crate registries (i.e., not on `crates.io`), then credentials should be set up in such a way that the Git command-line is able to clone the registry index.
Third-party crate registries usually provide instructions to achieve this.

Both of these are currently only possible when running Renovate self-hosted.

## Private sparse index registries

> If dependencies are available via a "sparse index", you may use
> `registryAliases` to let Renovate accesses the dependencies via the `sparse`
> protocol. If needed, set the authentication details with the `hostRules` config
> option. `crates.io` index access will still be via Renovates shared clone of
> the public Git index.

Reviewer: ```suggestion
If dependencies are available via a "sparse index", you may use `registryAliases` to let Renovate accesses the dependencies via the `sparse` protocol.
If needed, set the authentication details with the `hostRules` config option.
`crates.io` index access will still be via Renovate's shared clone of the public Git index.
```

We use the "one sentence per line" pattern in our docs. 😉

## Private sparse index registries

> If dependencies are available via a "sparse index", you may use
> `registryAliases` to let Renovate accesses the dependencies via the `sparse`
> protocol. If needed, set the authentication details with the `hostRules` config
> option. `crates.io` index access will still be via Renovates shared clone of the
> public Git index.

Reviewer: ```suggestion
If dependencies are available via a "sparse index", you may use `registryAliases` to let Renovate accesses the dependencies via the `sparse` protocol.
If needed, set the authentication details with the `hostRules` config option.
 `crates.io` index access will still be via Renovates shared clone of the public Git index.
```

For instance with a username and password:

> ```

Reviewer: ```suggestion
```json
```

"hostRules": [
    {
      "hostType": "cargo-git",
      "matchHost": "https://INSTANCE.jfrog.io",
      "username": "USERNAME",
      "password": "PASSWORD"
    },
    {
      "hostType": "cargo-http",
      "matchHost": "https://INSTANCE.jfrog.io",
      "authType": "Token-Only",
      "token": "Basic username-and-password-base64-encoded"
    }
  ]
```

With a bearer token:

> ```

Reviewer: ```suggestion
```json
```
